src/RileyMaze/RMZUnit.js

```diff
@@ -0,0 +1,219 @@
+            Gtk.Image, 'resource');
+        this._variablesCodeview.connect('should-compile', this._compile.bind(this));
+    }
+
+    bindGlobalModel(model) {
+        this._global = model;
+        this._globalNotifyHandler = model.connect('notify', this._onGlobalNotify.bind(this));
+
+        this._currentLevel = model.currentLevel;
+        this._bindLevelModel(this._global.getModel(this._currentLevel));
+    }
+
+    unbindGlobalModel() {
+        this._unbindLevelModel();
+        if (this._global && this._globalNotifyHandler) {
+            this._global.disconnect(this._globalNotifyHandler);
+            this._globalNotifyHandler = null;
+        }
+    }
+
+    _onGlobalNotify() {
+        if (this._global.currentLevel === this._currentLevel)
+            return;
+        this._currentLevel = this._global.currentLevel;
+        this._bindLevelModel(this._global.getModel(this._currentLevel));
+    }
+
+    _unbindLevelModel() {
+        if (this._model) {
+            if (this._notifyHandler) {
+                this._model.disconnect(this._notifyHandler);
+                this._notifyHandler = null;
+            }
+            if (this._bindings) {
+                this._bindings.forEach(binding => binding.unbind());
+                this._bindings = null;
+            }
+            this._model = null;
+        }
+    }
+
+    _bindLevelModel(model) {
+        this._unbindLevelModel();
+
+        this._model = model;
+        this._robotAUp.connect('toggled',
+            this._onActiveChanged.bind(this, 'robotADirection'));
+        this._model.connect('notify::robotADirection', () => {
+            const dirBool = directionStr[this._model.robotADirection];
+            if (dirBool === true)
+                this._robotAUp.set_active(true);
+            else if (dirBool === false)
+                this._robotADown.set_active(true);
+        });
+        this._robotBUp.connect('toggled',
+            this._onActiveChanged.bind(this, 'robotBDirection'));
+        this._model.connect('notify::robotBDirection', () => {
+            const dirBool = directionStr[this._model.robotBDirection];
+            if (dirBool === true)
+                this._robotBUp.set_active(true);
+            else if (dirBool === false)
+                this._robotBDown.set_active(true);
+        });
+        this._notifyHandler = model.connect('notify', this._onNotify.bind(this));
+        this._regenerateCode();
+    }
+
+    _onActiveChanged(childName, button) {
+        this._model.set_property(childName, directions[button.get_active()]);
+    }
+
+    bindWindow(win) {
+        win.lockscreen.key = 'item.key.rileymaze.1';
+        win.lockscreen.lock = 'lock.rileymaze.1';
+        void this;
+    }
+
+    _compile() {
+        const code = this._variablesCodeview.text;
+
+        if (code === '')
+            return;
+
+        const scope = {
+            robotADirection: null,
+            robotBDirection: null,
```

Comment:
For `up` and `down` to work as variables instead of string constants, they will have to be defined in the scope object. Currently if I choose "up" or "down" in the UI and let the code be regenerated, then it looks like it works initially; but if I add a space or change something else so it recompiles, then I get an error: "up is not defined".